File: 1_A_CSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace _1_A_CSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }


        private void button1_Click(object sender, EventArgs e)
        {
            // Clear richTextBox
            this.textBox.Text = "";
        }

        private void putTextButton_Click(object sender, EventArgs e)
        {
            // Insert text in the box
            this.textBox.Text = "A simple string :)";
        }

        private void textBox_MouseEnter(object sender, EventArgs e)
        {
            // Choose a random color each time the mouse enters the box
            Random rnd = new Random();
            Color randomColor = Color.FromArgb(rnd.Next(256), rnd.Next(256), rnd.Next(256));
            this.textBox.BackColor = randomColor;
        }

        private void textBox_MouseLeave(object sender, EventArgs e)
        {
            // Reset original color when mouse leaves
            this.textBox.BackColor = Color.White;
        }

        private void listBox1_DragEnter(object sender, DragEventArgs e)
        {
            if (e.Data.GetDataPresent(DataFormats.FileDrop))
                e.Effect = DragDropEffects.All;
            else
                e.Effect = DragDropEffects.None;
        }

        private void listBox1_DragDrop(object sender, DragEventArgs e)
        {
            string[] s = (string[])e.Data.GetData(DataFormats.FileDrop, false);
            int i;
            for (i = 0; i < s.Length; i++)
                listBox1.Items.Add(s[i]);
            
        }
    }
}

File: 1_A_VBNet/Form1.vb
Public Class Form1

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.AllowDrop = True
    End Sub
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Me.RichTextBox1.Text = "A Sample Text :)"
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Me.RichTextBox1.Text = ""
    End Sub

    Private Sub RichTextBox1_MouseEnter(sender As Object, e As EventArgs) Handles RichTextBox1.MouseEnter
        Dim rnd As New Random
        Me.RichTextBox1.BackColor = Color.FromArgb(255, rnd.Next(255), rnd.Next(255), rnd.Next(255))
    End Sub

    Private Sub RichTextBox1_MouseLeave(sender As Object, e As EventArgs) Handles RichTextBox1.MouseLeave
        Me.RichTextBox1.BackColor = Color.White
    End Sub

    Private Sub ListBox1_DragEnter(sender As Object, e As DragEventArgs) Handles ListBox1.DragEnter
        'If it contains a String type
        If e.Data.GetDataPresent(DataFormats.FileDrop) Then
            'Copy
            e.Effect = DragDropEffects.Copy
        Else
            'Otherwise undo the action
            e.Effect = DragDropEffects.None
        End If
    End Sub

    Private Sub ListBox1_DragDrop(sender As Object, e As DragEventArgs) Handles ListBox1.DragDrop
        'Get strings in memory
        Dim S() As String = e.Data.GetData(DataFormats.FileDrop)
        'Put the file's path in the list box
        For Each path In S
            Me.ListBox1.Items.Add(path)
        Next

    End Sub


End Class
